python-package/lightgbm/sklearn.py

```diff
@@ -2,23 +2,52 @@
 """Scikit-learn wrapper interface for LightGBM."""
 import copy
 from inspect import signature
-from typing import Callable, Dict, Optional, Union
+from typing import Callable, Dict, List, Optional, Tuple, Union
 
 import numpy as np
+import scipy.sparse as ss
 
 from .basic import Dataset, LightGBMError, _choose_param_value, _ConfigAliases, _log_warning
 from .callback import log_evaluation, record_evaluation
 from .compat import (SKLEARN_INSTALLED, LGBMNotFittedError, _LGBMAssertAllFinite, _LGBMCheckArray,
                      _LGBMCheckClassificationTargets, _LGBMCheckSampleWeight, _LGBMCheckXY, _LGBMClassifierBase,
                      _LGBMComputeSampleWeight, _LGBMLabelEncoder, _LGBMModelBase, _LGBMRegressorBase, dt_DataTable,
-                     pd_DataFrame)
+                     pd_DataFrame, pd_Series)
 from .engine import train
 
+_ArrayLike = Union[List, np.ndarray, pd_Series]
+_EvalResultType = Tuple[str, float, bool]
+
+_LGBM_ScikitCustomObjectiveFunction = Union[
+    Callable[
+        [_ArrayLike, np.ndarray],
+        Tuple[np.ndarray, np.ndarray]
+    ],
+    Callable[
+        [_ArrayLike, np.ndarray, np.ndarray],
```

Comment:
Shall we allow the users to pass `list` or `pd_Series` to the second parameter `y_pred` and the third parameter `group`, by changing the second and third argument type to `_ArrayLike`?

Reply:
Sorry for missing your discussion with @StrikerRUS above. After reading the comments, I agree with the current types for these parameters.